client/blocks/comment-button/index.jsx

```diff
@@ -20,51 +19,57 @@ import { getPostTotalCommentsCount } from 'calypso/state/comments/selectors';
 import './style.scss';
 
 function CommentButton( props ) {
-	const { commentCount, href, onClick, showLabel, tagName, target } = props;
+	const { commentCount, href, onClick, size, target } = props;
 	const translate = useTranslate();
 
-	return React.createElement(
-		tagName,
-		omitBy(
-			{
-				className: 'comment-button',
-				href: 'a' === tagName ? href : null,
-				onClick,
-				target: 'a' === tagName ? target : null,
-			},
-			( prop ) => prop === null
-		),
-		<Gridicon icon="comment" size={ props.size } className="comment-button__icon" />,
-		<span className="comment-button__label">
-			{ commentCount > 0 && <span className="comment-button__label-count">{ commentCount }</span> }
-			{ showLabel && commentCount > 0 && (
-				<span className="comment-button__label-status">
-					{ translate( 'Comment', 'Comments', {
-						context: 'noun',
-						count: commentCount,
-					} ) }
-				</span>
-			) }
-		</span>
+	return (
+		<a className="comment-button" href={ href } onClick={ onClick } tabIndex="0" target={ target }>
+			<Gridicon aria-hidden="true" icon="comment" size={ size } className="comment-button__icon" />
+			<span className="comment-button__label">
+				{ commentCount > 0 && (
+					<Fragment>
+						{ translate(
+							'{{count}}%(number)d{{/count}} {{label}}Comment{{/label}}',
```

Comment:
:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?

Reply:
Along with this, will this line be able to handle reordering it to "Comments 4" if a language requires it? I _think_ it can since it's all within one translate string but I can't remember for sure.

Reply:
Yes 👍 This wouldn't have worked previously, so it's an i18n improvement too :)